producers/fake.go

```diff
@@ -104,16 +104,16 @@ func (a *fakeProducer) generateEndpoint() (*pkg.Endpoint, error) {
 	}
 
 	switch a.mode {
-	case ipMode:
+	case "ip":
 		endpoint.IP = net.IPv4(
 			byte(randomNumber(1, 255)),
 			byte(randomNumber(1, 255)),
 			byte(randomNumber(1, 255)),
 			byte(randomNumber(1, 255)),
 		).String()
-	case hostnameMode:
+	case "hostname":
 		endpoint.Hostname = fmt.Sprintf("%s.%s", randomString(6), a.targetDomain)
-	case fixedMode:
+	case "fixed":
```

Comment:
you still have those modes defined as `const` up there. We should use them. Also I think it's a good idea.